superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricEditPopover/index.jsx

```diff
@@ -388,7 +388,7 @@ export default class AdhocMetricEditPopover extends React.PureComponent {
                       tabIndex={0}
                       role="button"
                       onClick={() => {
-                        this.props.handleDatasetModal(true);
+                        window.location.href = '/dataset/add/';
```

Comment:
Ditto above about this being a virtual dataset creation action.

Reply:
Confirmed: virtual only

Reply:
Fixed in [`this commit`](https://github.com/apache/superset/pull/22610/commits/d2903e307228270c40270088b7a83d64ae13e545).